scipy/signal/spectral.py

```diff
@@ -511,11 +511,429 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
     if noverlap is None:
         noverlap = nperseg // 8
 
-    freqs, time, Pxy = _spectral_helper(x, x, fs, window, nperseg, noverlap,
+    freqs, time, Sxx = _spectral_helper(x, x, fs, window, nperseg, noverlap,
                                         nfft, detrend, return_onesided, scaling,
-                                        axis, mode=mode)
+                                        axis, mode='psd')
+
+    return freqs, time, Sxx
+
+
+def check_COLA(window, nperseg, noverlap, tol=1e-10):
+    """
+    Check whether the Constant OverLap Add (COLA) constraint is met
+
+    Parameters
+    ----------
+    window : str or tuple or array_like
+        Desired window to use. See `get_window` for a list of windows and
+        required parameters. If `window` is array_like it will be used
+        directly as the window and its length must match nperseg.
+    nperseg : int
+        Length of each segment.
+    noverlap : int
+        Number of points to overlap between segments.
+    tol : float, optional
+        The allowed variance of a bin's weighted sum from the median bin
+        sum.
+
+    Returns
+    -------
+    verdict : bool
+        `True` if chosen combination satisfies COLA within `tol`, `False`
+        otherwise
+
+    See Also
+    --------
+    stft: Short Time Fourier Transform
+    istft: Inverse Short Time Fourier Transform
+
+    Notes
+    -----
+    In order to enable inversion of an STFT via the inverse STFT in
+    `istft`, the signal windowing must obey the constraint of "Constant
+    OverLap Add" (COLA). This ensures that every point in the input data is
+    equally weighted, thereby avoiding aliasing and allowing full
+    reconstruction.
+
+    Some examples of windows that satisfy COLA:
```

Comment:
There are actually more for each case:

- Hann at 1/2, 2/3, 3/4, 4/5, ...
- Bartlett at 1/2, 3/4, 5/6, 7/8, ...
- Blackman at 2/3, 4/5, 6/7, ...

```
check_COLA(hann(300, False), 300, 200)
Out[61]: True

check_COLA(bartlett(800, False), 800, 700)
Out[57]: True

check_COLA(blackman(700, False), 700, 600)
Out[75]: True
```

https://holometer.fnal.gov/GH_FFT.pdf has graphs of "Amplitude Flatness (AF)" for several windows, and the points where it peaks at 1 are the COLA conditions, if I understand correctly.